### PR TITLE
FTP-1380: Creation of Icon component, type and story.

### PR DESCRIPTION
**Pull request by subhasishnathAxelerant**
1380-pr

Screenshot 2022-09-30 at 10 24 23 AM

https://axelerant.atlassian.net/browse/FTP-1380

### CODE REVIEW

**Comment by subhasishnathAxelerant** [comment]
Lint is failing on pull. Which is fixed in https://github.com/axelerant-trainings-frontend/grocery-plus/pull/30 by Tushar.

File: components/Icon/Icon.test.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+describe('Test Icon component', () => {
+  it('renders the proper icon as passed down as props', () => {
+    render(
+      <Icon icon={HomeIcon} height="50" width="50" className="text-charcoal" />,
+    );
+    const svgIconPath = screen.getByTestId('svgIconPath');
+    expect(svgIconPath).toHaveAttribute(
+      'd',
+      'M8.19785 10.655V15.655C8.19785 16.205 7.74785 16.655 7.19785 16.655H3.19785C2.64785 16.655 2.19785 16.205 2.19785 15.655V8.655H0.497852C0.037852 8.655 -0.172148 8.085 0.167852 7.785L8.52785 0.255C8.90785 -0.085 9.48785 -0.085 9.86785 0.255L18.2279 7.785C18.5779 8.085 18.3579 8.655 17.8979 8.655H16.1979V15.655C16.1979 16.205 15.7479 16.655 15.1979 16.655H11.1979C10.6479 16.655 10.1979 16.205 10.1979 15.655V10.655H8.19785ZM14.1979 7.06768C14.1979 6.92594 14.1377 6.79086 14.0323 6.69603L9.53233 2.64603C9.34218 2.4749 9.05352 2.4749 8.86337 2.64603L4.36337 6.69603C4.25801 6.79086 4.19785 6.92594 4.19785 7.06768V14.155C4.19785 14.4311 4.42171 14.655 4.69785 14.655H5.69785C5.97399 14.655 6.19785 14.4311 6.19785 14.155V8.655H12.1979V14.155C12.1979 14.4311 12.4217 14.655 12.6979 14.655H13.6979C13.974 14.655 14.1979 14.4311 14.1979 14.155V7.06768Z',
```
Comment (skippednote) [inline components/Icon/Icon.test.tsx:44]: Please group all the tests in a single describe block

File: components/Icon/Icon.test.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { render, screen } from '@testing-library/react';
+import Icon from './Icon';
+import { HomeIcon, DashIcon, ArrowForwardIcon, BagIcon } from '../iconLibrary';
+
+describe('Test Icon component', () => {
```
Comment (skippednote) [inline components/Icon/Icon.test.tsx:5]:
```suggestion
describe('Icon', () => {
```

File: @types/Icon.types.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+type iconType = {
+  icon: string;
+  height: string;
+  width: string;
+  className: string;
```
Comment (skippednote) [inline @types/Icon.types.ts:5]:
```suggestion
  className: 'charcoal' | 'white';
```

File: components/Icon/Icon.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import iconType from '../../@types/Icon.types';
+
+const Icon: React.FC<iconType> = ({ icon, width, height, className = '' }) => {
```
Comment (skippednote) [inline components/Icon/Icon.tsx:4]:
```suggestion
const Icon: React.FC<iconType> = ({ icon, width, height, variant }) => {
```

File: components/Icon/Icon.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <div
+      className={`icon w-fit h-auto ${className}`}
```
Comment (skippednote) [inline components/Icon/Icon.tsx:7]:
```suggestion
      className={`icon w-fit h-auto text-${variant}`}
```

File: components/Icon/Icon.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    >
+      <svg
+        width={width}
+        height={height}
+        viewBox="0 0 19 21"
+        fill="none"
+        xmlns="http://www.w3.org/2000/svg"
+        data-testid="svgIcon"
+      >
+        <path d={icon} fill="currentcolor" data-testid="svgIconPath" />
```
Comment (skippednote) [inline components/Icon/Icon.tsx:18]: use hyphen instead of lowerCamelCase

File: components/Icon/Icon.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+      className={`icon w-fit h-auto ${className}`}
+      data-testid="svgIconWrapper"
+    >
+      <svg
+        width={width}
+        height={height}
+        viewBox="0 0 19 21"
+        fill="none"
+        xmlns="http://www.w3.org/2000/svg"
+        data-testid="svgIcon"
```
Comment (skippednote) [inline components/Icon/Icon.tsx:16]: use hyphen instead of lowerCamelCase

**Review by skippednote — CHANGES_REQUESTED** [review]

File: components/Icon/Icon.test.tsx (hunk 2 above)
Comment (subhasishnathAxelerant) [reply components/Icon/Icon.test.tsx:5]: Done.

File: @types/Icon.types.ts (hunk 3 above)
Comment (subhasishnathAxelerant) [reply @types/Icon.types.ts:5]: Done.

File: components/Icon/Icon.test.tsx (hunk 1 above)
Comment (subhasishnathAxelerant) [reply components/Icon/Icon.test.tsx:44]: Done.

File: @types/Icon.types.ts (hunk 3 above)
Comment (subhasishnathAxelerant) [reply @types/Icon.types.ts:5]: Done.

File: components/Icon/Icon.tsx (hunk 5 above)
Comment (subhasishnathAxelerant) [reply components/Icon/Icon.tsx:7]: Changed to:
`className={`icon w-fit h-auto text-${variant} ${extraClasses ? extraClasses : ""}`}`

File: components/Icon/Icon.tsx (hunk 7 above)
Comment (subhasishnathAxelerant) [reply components/Icon/Icon.tsx:16]: Done.

File: components/Icon/Icon.tsx (hunk 6 above)
Comment (subhasishnathAxelerant) [reply components/Icon/Icon.tsx:18]: Done

**Review by skippednote — APPROVED** [review]

**Pull request merged**